corehq/apps/data_interfaces/forms.py

```diff
@@ -828,6 +837,22 @@ def clean_property_match_definitions(self):
                     'property_value': str(property_value),
                     'match_type': match_type,
                 })
+            elif match_type == MatchPropertyDefinition.MATCH_REGEX:
+                property_value = obj['property_value']
+
+                if not property_value:
+                    raise ValidationError(_("Please enter a valid regular expression to match"))
+
+                try:
+                    re.compile(property_value)
+                except:
```

Comment:
Bare `except` is not recommended because it can catch errors that this code should not attempt to recover from or silence (`KeyboardInterrupt`, `SystemExit`). Better to use `except Exception:`, although even that is overly broad because it will catch things like `MemoryError` which probably cannot be recovered from here. Would it work to use `except (re.error, ValueError, TypeError):`?

Aside (not your problem): `re.error` should probably extend `ValueError` instead of `Exception`. One thing I really miss from Java is documentation of what exceptions might possibly be raised by a given function. Python standard library documentation, while very good, is notoriously bad at detailing that part of the interface.

This comment also applies uses of bare `except` in data_interfaces/models.py

Reply:
good points, updated here https://github.com/dimagi/commcare-hq/pull/20824